### DISCUSSION TITLE
How to configure `exports:` block in Operator when using fully-managed GMP?

### DISCUSSION THREAD

**Discussion by nategoethel** [GoogleCloudPlatform/prometheus-engine#1171] [Q&A]
https://github.com/GoogleCloudPlatform/prometheus-engine/pull/884 added the ability to add URLs to the `exports` key in the Operator configuration to allow metrics to be written to other backends (e.g. Coralogix in my case) as well as the main Cloud Monitoring backend. However, we have configured [managed collection](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-managed#enable-mgdcoll-gke), which means we did not install the Operator and other CRDs ourselves, and thus cannot edit the Operator to specify this key. Is there a way we can still do this?

**Comment by lyanco** [comment]
You need to edit the OperatorConfig, not the operator code. `kubectl -n gmp-public edit operatorconfig config` should work.

**Reply by nategoethel** [reply]
Hmm, OK. Something like this? 

```
apiVersion: monitoring.googleapis.com/v1
exports:
- url: https://ingress.cx498.coralogix.com/prometheus/v1
kind: OperatorConfig
managedAlertmanager:
  configSecret:
    key: alertmanager.yaml
    name: alertmanager
metadata:
  annotations:
    components.gke.io/component-name: managed-prometheus
    components.gke.io/component-version: 0.8.3-gke.0
    components.gke.io/layer: addon
  creationTimestamp: "2024-03-29T16:21:13Z"
  generation: 2
  labels:
    addonmanager.kubernetes.io/mode: EnsureExists
  name: config
  namespace: gmp-public
  resourceVersion: "242122378"
  uid: 89e2841d-71b4-4c5f-bfa0-0d3424a18d87
```

**Reply by nategoethel** [reply]
I'm also wondering if there's a way to pass a bearer token? For example, with "regular" prometheus, we would do something like this, as the remote endpoint requires authentication:
```
 ## Remote write configs for Coralogix
  secrets: ['coralogix-keys']
  remoteWrite:
    - url: "https://ingress.cx498.coralogix.com/prometheus/v1"
      name: 'crx'
      remoteTimeout: 120s
      bearerTokenFile: '/etc/prometheus/secrets/coralogix-keys/PRIVATE_KEY'
```

**Reply by lyanco** [reply]
Yup.

We don't have any way to specify PRW options, sorry. It'll be supported once we unfork our collector, expected mid-next year.
